addons/udes_sale_stock/models/sale_order_line.py

```diff
@@ -144,28 +142,36 @@ def _set_ui_is_cancelled(self):
         lines_to_uncancel = self.filtered(lambda l: l.is_cancelled and not l.ui_is_cancelled)
         lines_to_cancel = self.filtered(lambda l: not l.is_cancelled and l.ui_is_cancelled)
 
-        # Check if the warehouse config allow manual cancellation
-        if lines_to_cancel and False in self.mapped(
-            "order_id.warehouse_id.u_allow_manual_sale_order_line_cancellation"
-        ):
-            raise ValidationError(
-                _(
-                    "Manual cancellation of individual order lines is not "
-                    "allowed by the warehouse config"
-                )
-            )
-
         # Forbid uncancellation of sale order lines
         if lines_to_uncancel:
             raise ValidationError(
                 _("Cannot uncancel order lines: %s")
                 % (", ".join(lines_to_uncancel.mapped("name")),)
             )
 
+        cancel_errors = lines_to_cancel._get_cancel_errors()
+        if cancel_errors:
+            # Just raise for the first error in the list (this preserves behaviour
+            # from before we had a helper _get_cancel_errors())
+            raise ValidationError(cancel_errors[0])
+
+        lines_to_cancel.action_cancel()
+
+    def _get_cancel_errors(self):
+        """Retrieve any errors for cancelling the sale line"""
+        errors = []
+        # Check if the warehouse config allow manual cancellation
```

Comment:
Nit :)
```suggestion
        # Check if the warehouse config allows manual cancellation
```

Reply:
7865144

I also dropped the `moves` variable :)